jetty-servlets/src/main/java/org/eclipse/jetty/servlets/CrossOriginFilter.java

```diff
@@ -501,11 +511,41 @@ private String commify(List<String> strings)
         return builder.toString();
     }
 
+    /** 
+     * creates a pattern to match any of a list of  string
+     * If the input list is ["foo, "bar"] it will generate
+     * a Pattern for the regex "(\\Efoo\\E|\\Ebar\\E) 
+     */
+    static Pattern patternize(final List<String> strings) 
+    {
+        final StringBuilder builder = new StringBuilder();
+        for (int i = 0; i < strings.size(); ++i)
+        {
+            final String string = strings.get(i);
+            if (string.trim().isEmpty()) break;
```

Comment:
`String.trim().isEmpty()` is wasteful creation of a new object to test a state to then just throw it out..
Either consider Storing the trimmed string for later use (like @Maaartinus points out below)
Or use `StringUtil.isBlank(string)` from the `org.eclipse.jetty.util` package if the trimmed string isn't relevant below.

Reply:
You are right. I was looking the gains for N request that I forget to be efficient for the one call of the method.

I also changed how the loop is made. I took the style from the commify but I see that for-each can be used.